### PR TITLE
feat(studio): save an experiment's column layout (ASE-425)

### PR DESCRIPTION
**Pull request by rrhyne**
https://github.com/user-attachments/assets/d43069e8-57e5-4e8a-89bd-fc09044597c7

## Summary

The evaluations table's column order lived in `localStorage`, so it followed the browser rather than the experiment it describes — which of an experiment's evaluators are worth a column, and in what order they read, is a property of the experiment, not of the machine looking at it. Column visibility was not persisted at all. Both now live on the experiment: a **Save columns** button appears at the toolbar's leading edge once the live layout differs from the saved one, and disappears again once it matches.

## Changes

- `ExperimentGroup` gains `column_layout` (`order` + `hidden`). The entity store is schema-on-read, so **no migration is required**. Update follows the existing `pareto` convention: an omitted `column_layout` preserves what is saved rather than clearing it.
- The field is **nullable** so "never saved" stays distinct from "saved with nothing hidden". The client applies its own default hidden columns only in the null case, so a deliberate "show everything" survives a reload instead of being re-hidden — and the Save button does not appear on a freshly loaded page.
- Visibility is stored as the **hidden ids** rather than a map over every column, so a column that appears later (a new evaluator, a new metadata key) shows up by default instead of being invisible because a layout saved before it existed never mentioned it.
- Studio seeds column order and visibility from the experiment and persists them through the existing `PUT /experiments/{name}`.
- `StudioDataViewToolbar` gains a `slotStart` (surfaced as `toolbarSlotStart` on `StudioDataView`), rendered before the search bar. Additive and optional; existing callers are unaffected.
- Regenerated `openapi/`, the web SDK, and `plugin-types/plugin.d.ts`.

### Behavior change worth calling out

This replaces the `localStorage` column-order persistence (`nemo-studio:experiment-columns:<id>`). Anyone holding a local layout will see it reset once to the experiment's saved (initially absent) layout. Keeping both stores seemed worse than picking one.

## Type of Change

- [x] Code change (feature, bug fix, or refactor)
- [ ] Code change with documentation updates
- [ ] Documentation only
- [ ] Contributor tooling or automation
- [ ] CI, build, or test infrastructure

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [ ] Documentation updated for user-visible behavior
- [x] Documentation not applicable — justification: no documented surface changes; the field is described in the OpenAPI schema it ships with.

## Verification

- [x] Pull request title follows the repository's Conventional Commit format
- [x] Every commit includes an appropriate `Signed-off-by:` trailer
- [ ] `uv run pre-commit run -a` passes, or any blocked checks are identified below
- [x] Targeted tests pass, or tests are marked not applicable above
- [x] No secrets, API keys, or credentials are included

Targeted validation, all run on the current base:

- `uv run pre-commit run --files <changed files>` — passed (ruff, ruff format, **ty**, copyright headers, UI lint-staged, merge-conflict check). The full `-a` run was **not** performed locally, so that box is left unchecked; CI runs the complete gate.
- `uv run --frozen pytest services/intake/tests/integration/test_experiments_crud.py -k "column_layout or pareto"` — 3 passed. Covers the null default, the PUT round-trip, preservation on an update that omits the field, and that an empty layout stays distinguishable from null.
- `pnpm --filter="...[origin/main]" run --parallel --if-present test:ci` — `@nemo/common` 1518 passed (122 files), `studio` 3358 passed (346 files). Includes 11 new `useColumnLayout` cases (load-time clean state, hide/show, reorder, hidden-compared-as-a-set).
- `pnpm --filter="...[origin/main]" run --parallel --if-present typecheck`, `pnpm lint` — clean.
- Re-ran both generators on the current base and confirmed no drift: the committed OpenAPI spec and `plugin.d.ts` match what the generators produce.
- Manually exercised against a local platform: reordering and hiding columns reveals the button, saving persists, and a reload comes back on the saved layout. The stored layout round-trips through the API.

Not done: the Stainless **Python** SDK was not regenerated — `make update-sdk` requires `STAINLESS_API_KEY`, which this environment does not have. The OpenAPI spec and web SDK are committed.

## Summary by CodeRabbit

- **New Features**
  - Experiment evaluation tables now save and restore customized column order and visibility.
  - Save layouts from the table toolbar with clear saving feedback.
  - Data view toolbars can display custom content at the leading edge.
  - Experiment APIs support optional saved column layouts.

- **Bug Fixes**
  - Preserved the distinction between no saved layout and an intentionally empty layout.
  - Partial experiment updates now preserve fields omitted from requests.
  - Column state no longer carries over when navigating between experiments.

- **Tests**
  - Added coverage for layout persistence, updates, reordering, visibility changes, and unsaved-state detection.

### CODE REVIEW

File: web/packages/studio/src/components/dataViews/ExperimentDataView/useColumnLayout.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      workspace,
+      name: group.name,
+      data: {
+        name: group.name,
+        description: group.description,
+        insight_id: group.insight_id,
+        summary: group.summary,
+        metadata: group.metadata,
+        default_sort: group.default_sort,
+        column_layout: { order: columnOrder, hidden: currentHidden },
```
[the remark this answers is not in the document]
Comment (rrhyne) [reply web/packages/studio/src/components/dataViews/ExperimentDataView/useColumnLayout.ts]: Fixed in d34393639, at the endpoint rather than the client. The API blanked `description`/`insight_id`/`summary`/`metadata` when omitted, which is why the client echoed them back; they're now guarded on `model_fields_set` like every field added since, and explicit null still clears. The layout save sends only `column_layout`. Preferred over optimistic concurrency because it removes the need to send those fields at all rather than detecting the collision that sending them creates. Test added covering preservation and null-clears.

File: web/packages/studio/src/components/dataViews/ExperimentDataView/index.tsx [hunk 2]
```diff
@@ -179,33 +182,30 @@ export const ExperimentDataView: FC<ExperimentDataViewProps> = ({
...
     multiSort: true,
-    columnVisibility: { created_by: false, updated_at: false },
+    columnVisibility: seededColumns.columnVisibility,
     // Keep the pin toggle, row selection, and the row actions menu reachable while horizontally
     // scrolling this wide table.
     columnPinning: { left: ['pin', 'row-selection'], right: ['row-actions'] },
     filterFieldMap: getEvaluationFilterField,
-    columnOrder: savedColumnOrder ?? [],
+    columnOrder: seededColumns.columnOrder,
   });
```
[the remark this answers is not in the document]
Comment (rrhyne) [reply web/packages/studio/src/components/dataViews/ExperimentDataView/index.tsx:201]: Fixed in d34393639 by keying `ExperimentDataView` with `group.id` at its mount site in `ExperimentDetailRoute` — the same pattern the Pareto and trend charts in this file already use for exactly this reason (the route doesn't remount between experiments).

Skipped the rerender test: with the fix at the call site, a test that swaps the `group` prop would assert React's key semantics rather than our logic, and the layout derivation it would exercise is already covered by the `useColumnLayout` unit tests.

File: openapi/ga/individual/platform.openapi.yaml [hunk 3]
```diff
@@ -12189,6 +12222,12 @@ components:
           title: Default Sort
         pareto:
           $ref: '#/components/schemas/ParetoConfig'
+        column_layout:
+          allOf:
+          - $ref: '#/components/schemas/ColumnLayout'
+          description: Saved column order and column visibility for the experiment's
+            evaluations table. Null when no layout has been saved, which is distinct
+            from a saved layout that hides nothing.
```
[the remark this answers is not in the document]
Comment (rrhyne) [reply openapi/ga/individual/platform.openapi.yaml:12235]: Fixed in d34393639, but at the source: `openapi/` is generated by `make refresh-openapi`, so a hand-edit there is overwritten. Added `json_schema_extra={"nullable": True}` to the pydantic field (the mechanism `baseline_evaluation_name` already uses) and regenerated — the response now carries `nullable: true` and the TS client is `column_layout?: ColumnLayout | null`.

Applied to the response only: on the request, null and omitted are equivalent ("no change"), so marking it nullable there would imply null clears the layout, which it doesn't.

Worth noting it caught a real bug — the honest type broke `tsc` and revealed the seeding helpers only accepted `undefined` while the API has always been able to send null.

File: web/packages/common/plugin-types/plugin.d.ts [hunk 4]
```diff
@@ -9316,6 +9318,7 @@ interface Props$4<DataType> {
    * Rendered at the trailing end of the toolbar row (after the search bar and filter toggle).
    * Useful for controls like a sort dropdown that belong visually in the toolbar.
    */
+  toolbarSlotStart?: ReactNode;
```
[the remark this answers is not in the document]
Comment (rrhyne) [reply web/packages/common/plugin-types/plugin.d.ts]: Good catch, my insertion error. Fixed in 35f67cf9f at the source (`StudioDataView.tsx`; `plugin.d.ts` is generated): the trailing-content doc is back on `toolbarSlotEnd` and `toolbarSlotStart` has its own one-liner. Regenerated the plugin surface.

**Review by aray12 — APPROVED** [review]

**Review by steramae-nvidia — APPROVED** [review]

**Review by htolentino-nvidia — APPROVED** [review]

**Comment by rrhyne** [comment]
Nice loader on the logs.

Breadcrumb is inconsistent between pages:
*  the navigation is Fine-tune 
* the title on the list is "Custom models"
* the create screen is "[New Fine-Tuned Model](https://ns.local.aire.nvidia.com:5173/workspaces/default)"
* the view page on a fine tuned model run is Customizations

Should we be graphing each point for validation loss? Perhaps a line without points that allows the user to roll over and see the value would be more readable?
